nimare/dataset/dataset.py

```diff
@@ -1,68 +1,187 @@
+
+    # Reduce images database according to additional criteria
+    # Only keep unthresholded, MNI, group level fMRI maps
+    red_df = image_df.loc[image_df['modality'] == 'fMRI-BOLD']
+    red_df = red_df.loc[red_df['image_type'] == 'statistic_map']
+    red_df = red_df.loc[red_df['analysis_level'] == 'group']
+    red_df = red_df.loc[red_df['is_thresholded'] is False]
+    red_df = red_df.loc[red_df['not_mni'] is False]
+
+    # Look for relevant metadata
+    red_df = red_df.dropna(subset=['cognitive_paradigm_cogatlas'])
+
+    ## MFX/FFX GLMs need contrast (beta) + standard error
+    mffx_df = red_df.loc[red_df['map_type'] == 'univariate-beta map']
 
-        if not isinstance(conn_matrices, list):
-            conn_matrices = [conn_matrices]
+    ## RFX GLMs need contrast (beta)
+    rfx_df = red_df.loc[red_df['map_type'] == 'univariate-beta map']
 
-        for conn_mat in conn_matrices:
-            if not isinstance(conn_mat, ConnMatrix) and conn_mat is not None:
-                raise ValueError('All conn_matrices inputs must be nimare ConnMatrices.')
-            elif isinstance(conn_mat, ConnMatrix):
-                self.connectomes[conn_mat.type] = conn_mat
+    ## Stouffer's, Stouffer's RFX, and Fisher's IBMAs can use Z maps.
+    # T and F maps can be transformed into Z maps, but T maps need sample size.
+    # Only keep test statistic maps
+    acc_map_types = ['Z map', 'T map', 'F map']
+    st_df = red_df.loc[red_df['map_type'].isin(acc_map_types)]
+    keep_idx = st_df['map_type'].isin(['Z map', 'F map'])
+    keep_idx2 = (st_df['map_type'] == 'T map') & ~pd.isnull(st_df['number_of_subjects'])
+    keep_idx = keep_idx | keep_idx2
+    st_df = st_df.loc[keep_idx]
 
-        for surf in surfaces:
-            if not isinstance(surf, Surface) and surf is not None:
-                raise ValueError('All conn_matrices inputs must be nimare ConnMatrices.')
-            elif surf.type in self.surfaces.keys():
-                self.surfaces[surf.type] = surf
+    ## Weighted Stouffer's IBMAs need Z + sample size.
+    st_df['id_str'] = st_df['image_id'].astype(str).str.zfill(6)
+
+    if not isdir(out_dir):
+        mkdir(out_dir)
+        api.download_images(out_dir, red_df, target=None, resample=False)
+    elif overwrite:
+        # clear out out_dir
+        raise Exception('Currently not prepared to overwrite database.')
+        api.download_images(out_dir, red_df, target=None, resample=False)
+
+
+class Database(object):
+    """
+    Storage container for a coordinate- and/or image-based meta-analytic
+    dataset/database.
+
+    Parameters
+    ----------
+    dataset_file : :obj:`str`
+        Json file containing dictionary with database information.
+    """
+    def __init__(self, database_file):
+        with open(database_file, 'r') as fo:
+            self.data = json.load(fo)
+
+    def get_dataset(self, search='', algorithm=None, target=None):
```

Comment:
If we push transforming images and, more importantly, coordinates until Dataset generation, then I don't think we can do searches using ROIs in the Database. At least the way I've been thinking about it, generally speaking one would only transform a subset of a Database into a Dataset, based on some criteria (e.g., necessary data for a specific estimator, tagged with some CogPO or CogAt label, etc.), which means searching the Database to get a list of Contrast IDs before feeding those into a `get_dataset` method (or something similar). However, without the coordinates all in the same space as some ROI, we can't search based on their locations (unless we transform on the fly in the search method, which I think would be wasteful). I'm fine with allowing metadata-based searching within Databases, but restricting location-based searching to Datasets, but it's probably something worth discussing.